src/server.js

```diff
@@ -91,20 +96,37 @@ app.use(
             createdVmContext: false,
         }: RequestStats);
 
+        /**
+         * Before we do any real work, we may have added the log middleware
+         * and we'll want everything to use it. Rather than pass that around
+         * or pass the request around, we can use our logging module and
+         * tell it to scope logging to the request logger during the request.
+         * NOTE: the $Request type doesn't have a log field, officially.
+         * $FlowIgnore
+         */
+        const requestLog: Logger = req.log;
+        if (requestLog != null) {
+            useScopedLogger(requestLog);
+        }
+
         pendingRenderRequests++;
         const renderProfile = profile.start("/render");
 
         // Register for the response finish so we can finish up our stats.
         res.on("finish", () => {
-            pendingRenderRequests--;
-            if (res.statusCode < 300) {
-                // only log on successful fetches
-                const renderBody: RenderBody = (req.body: any);
-                renderProfile.end(
-                    `render-stats for ${
-                        renderBody.urls[renderBody.urls.length - 1]
-                    }: ${JSON.stringify(res.locals.requestStats) || ""}`,
-                );
```

Comment:
Unchanged, just now wrapped by a try to make sure we revert the logging scope if for some reason, this threw an error.

Reply:
Irrelevant after rework.